prow/git/v2/client_factory.go

```diff
@@ -59,6 +59,8 @@ type repoClient struct {
 type ClientFactoryOpts struct {
 	// Host, defaults to "github.com" if unset
 	Host string
+	// Whether to use HTTP. Defaults to false (i.e., defaults to HTTPS).
+	Http *bool
 	// UseSSH, defaults to false
```

Comment:
Is this also an alternative to UseSSH? Probably should indicate the precedence between them, or consider changing to an enum.

Also wondering:
a) whether the "UseHttp" is more consistent, and canonical for a boolean?
b) whether it should be UseInsecureHTTP, to improve self-documentation?

Reply:
> Is this also an alternative to UseSSH?

Using HTTPS is an alternative to using SSH. Sadly "UseHTTPS" is not a field at all.

> Probably should indicate the precedence between them, or consider changing to an enum.

I think changing to an enum is desirable (have 3 values in there, HTTPS (default), HTTP, and SSH). But I think that might result in a noisy refactor. For now I will indicate in the comment that these are all mutually exclusive values.

> Also wondering:
> a) whether the "UseHttp" is more consistent, and canonical for a boolean?

I like that name better.

> b) whether it should be UseInsecureHTTP, to improve self-documentation?

Good idea, even better (because "HTTP" can be ambiguous by itself). Will update.

Reply:
Done.